### ISSUE TITLE
Segfault for only (for now) `tictactoe.b` with simple loop optimizations only

### ISSUE THREAD

**Issue by khagankhan** [khagankhan/brainflux#4]
Code:
```bf
--->--->>>>->->->>>>>-->>>>>>>>>>>>>>>>>>+>>++++++++++[
  <<++[
    --<+<<+<<+>>>>[
      >[<->>+++>>[-]+++<<<+[<++>>+<--]]+>+++++[>>+++++++++<<-]
      >>++++.[-]>>+[<<<<+>>+>>-]<<<<<<[>+<-]<<
    ]++++++++++.[-]>++
  ]-->>[-->[-]>]<<[
    >>--[
      -[
        -[
          -----[>+>+++++++<<+]-->>-.----->,[<->-]<[[<]+[->>]<-]<[<<,[-]]>>>>
        ]>
      ]<[
        >-[+<+++]+<+++[+[---->]+<<<<<<[>>]<[-]]
        >[<+[---->]++[<]<[>]>[[>]+>+++++++++<<-[<]]]>[>>>>]
      ]<[
        -[[>+>+<<-]>[<+>-]++>+>>]<[<<++[-->>[-]]>[[-]>[<<+>>-]>]]
      ]<[
        [[<<]-[>>]<+<-]>[-<+]<<[<<]-<[>[+>>]>[>]>[-]]
        >[[+>>]<-->>[>]+>>>]
      ]<[
        -[
          --[+<<<<--[+>[-]>[<<+>+>-]<<[>>+<<-]]++[>]]
          <<[>+>+<<-]>--[<+>-]++>>>
        ]<[<<<[-]+++>[-]>[<+>>>+<<-]+>>>]
      ]<[
        +[[<]<<[<<]-<->>+>[>>]>[>]<-]+[-<+]<++[[>+<-]++<[<<->>+]<++]<
        <<<<<<      +> > >+> > >+[
        <<<               ->+>+>+[
        <<<<<<<   +>->+> > >->->+[
        <<<<<         ->+>+> >+>+[
        <<<<            ->->+>->+[
        <<<<<<<<+>-> >+> > >->+>+[
        <<<<<         -> >+> >->+[
        <<<<            +>->+> >+]]]]]]]
        +++[[>+<-]<+++]--->>[-[<->-]<++>>]++[[<->-]>>]>[>]
      ]<
    ]
  ]<
]
```

With optimizations except simple loops: (Increment/Decrement Fusions and memory scannig for _[>]_ loops) works fine:

```
X23
456
789
>5
X23
4O6
78X
>6
X23
XOO
78X
>7
X2X
XOO
O8X
>3
```
With simple loop optimizations:

```
X23
456
789
�5
zsh: segmentation fault  ./tic
```
```

**Comment by khagankhan** [comment]
The culprit is that we skip the current cell by just zeroing it. However, "[<<->>+]" is not zeroing the current cell that is why it should be handled correctly. Tentatively: `pointer_change == 0 && (/*cell_modifications == 1 || */ cell_modifications == -1) && (loop_start == 1 && loop_end == 1)` I commented for `Z[(>0+1)]` cases and it worked perfectly. I will update the optimizations for that. Probably `ZeroAndModify` token will be updated to something like "SimpleLoop" token for better understanding.

**Comment by khagankhan** [comment]
No need for changing the name. I guess ZeroAndModify is correct naming. For zeroing with [+] we do not multiply and add/sub with the p[0] but 256 - p[0] because it takes not p[0] iterations to zero the current cell but 256 - p[0] iterations. Fixed by this commit: 2c0bc823060798d5e773a8dfcaea6d8684219cef
